global-search/src/modules/FilterModule.ts

```diff
@@ -0,0 +1,42 @@
+import {render} from 'lit-html';
+import {filterResultsTemplate} from '../template/filter-results';
+import {urlParameterMap} from '../utils/urlParameter';
+import {ParamMap} from '../types/url-parameters';
+
+export class FilterModule {
+    private urlParameter: ParamMap;
+
+    constructor() {
+        this.urlParameter = urlParameterMap();
+        this.setSessionStorage();
+        this.filterResultsTemplate();
+    }
+
+    /**
+     * setSessionStorage function set profile, scope and label from the URL parameters
+     * @return {undefined}
+     * */
+    setSessionStorage() {
+        let profile = this.urlParameter.profile;
+        let scope = this.urlParameter.scope
+        let label = this.urlParameter.label
+
+        if(profile && profile !== 'qld' || scope){
+            sessionStorage.setItem('fcProfile', profile);
+            sessionStorage.setItem('fcScope', scope);
+        }
+        label ? sessionStorage.setItem('fcLabel', label) : '';
+    }
+
+    /**
+     * filterResultsTemplate display a template to filter search results
+     * @return {undefined}
+     * */
+    filterResultsTemplate(){
+        let profile = this.urlParameter.profile;
+
+        if(this.urlParameter.filter || profile && profile !== 'qld' || this.urlParameter.scope){
+            render(filterResultsTemplate(), document.getElementById('qg-filter-by-results') as HTMLBodyElement);
```

Comment:
not sure if type assertion is needed here, maybe check `document.getElementById('qg-filter-by-results')` is not undefined first?